src/main/java/factionsystem/Objects/Gate.java

```diff
@@ -0,0 +1,717 @@
+					_bottomY = coord1.getY();
+				}
+				final int bottomY = _bottomY;
+				int _leftZ = coord1.getZ();
+				int _rightZ = coord2.getZ();
+	
+				if (coord2.getZ() < coord1.getZ())
+				{
+					_leftZ = coord2.getZ();
+					_rightZ = coord1.getZ();
+				}
+				final int leftZ = _leftZ;
+				final int rightZ = _rightZ;
+				
+				int c = 0;
+				for (int y = topY; y >= bottomY; y--)
+				{
+					c++;
+					final int blockY = y;
+					Bukkit.getScheduler().runTaskLater(main, new Runnable() {
+						Block b;
+	                    @Override
+	                    public void run() {
+	        				for (int z = leftZ; z <= rightZ; z++)
+	        				{
+	        					b = getWorld().getBlockAt(coord1.getX(), blockY, z);
+	        					b.setType(material);
+	        					b.getState().update(true);
+	        					getWorld().playSound(b.getLocation(), soundEffect, 0.1f, 0.1f);
+	        				}
+	        				if (blockY == bottomY + 1)
+	        				{
+	        					gateStatus = GateStatus.READY;
+	        				}
+	                    }
+	                }, c * 10);
+				}			
+			}
+		}
+		else
+		{
+			// TODO: Bridge code iterates over x/z
+		}
+	}
+	
+	public void fillGate()
+	{
+
+		if (!open)
+			return;
+		
+		open = false;
+		// For vertical we only need to iterate over x/y
+		if (vertical)
+		{
+			if (isParallelToX())
+			{
+				int topY = coord1.getY();
+				int bottomY = coord2.getY();
+				if (coord2.getY() > coord1.getY())
+				{
+					topY = coord2.getY();
+					bottomY = coord1.getY();
+				}
+				
+				int _leftX = coord1.getX();
+				int _rightX = coord2.getX();
+				if (coord2.getX() < coord1.getX())
+				{
+					_leftX = coord2.getX();
+					_rightX = coord1.getX();
+				}
+	
+				final int leftX = _leftX;
+				final int rightX = _rightX;
+				
+				for (int y = topY; y >= bottomY; y--)
+				{
+					Block b = null;
+    				for (int x = leftX; x <= rightX; x++)
+    				{
+    					b = getWorld().getBlockAt(x, y, coord1.getZ());
+    					b.setType(material);
+    				};
+					if (b != null)
+						getWorld().playSound(b.getLocation(), soundEffect, 0.1f, 0.1f);
+				}
+			}
+			else if (isParallelToZ())
+			{
+				int topY = coord1.getY();
+				int bottomY = coord2.getY();
+				if (coord2.getY() > coord1.getY())
+				{
+					topY = coord2.getY();
+					bottomY = coord1.getY();
+				}
+				
+				int _leftZ = coord1.getZ();
+				int _rightZ = coord2.getZ();
+	
+				if (coord2.getZ() < coord1.getZ())
+				{
+					_leftZ = coord2.getZ();
+					_rightZ = coord1.getZ();
+				}
+				final int leftZ = _leftZ;
+				final int rightZ = _rightZ;
+				
+				for (int y = topY; y >= bottomY; y--)
+				{
+					Block b = null;
+    				for (int z = leftZ; z <= rightZ; z++)
+    				{
+    					b = getWorld().getBlockAt(coord1.getX(), y, z);
+    					b.setType(material);
+    				};
+					if (b != null)
+						getWorld().playSound(b.getLocation(), soundEffect, 0.1f, 0.1f);
+				}		
+			}
+		}
+		else
+		{
+			// TODO: Bridge code iterates over x/z
```

Comment:
Does this need to be completed before merging?

Reply:
It only means there is no horizontal gate (IE: bridge) support right now, but I figured since the gates were the core requirement this would be more of a nice-to-have.

Reply:
I gotchya, that can totally be added later